pkg/controller/operators/olm/requirements.go

```diff
@@ -70,148 +69,57 @@ func (a *Operator) minKubeVersionStatus(name string, minKubeVersion string) (met
 	return
 }
 
-func (a *Operator) requirementStatus(strategyDetailsDeployment *v1alpha1.StrategyDetailsDeployment, crdDescs []v1alpha1.CRDDescription,
+func (a *Operator) requirementStatus(crdDescs []v1alpha1.CRDDescription,
 	ownedAPIServiceDescs []v1alpha1.APIServiceDescription, requiredAPIServiceDescs []v1alpha1.APIServiceDescription,
-	requiredNativeAPIs []metav1.GroupVersionKind) (met bool, statuses []v1alpha1.RequirementStatus) {
+	requiredNativeAPIs []metav1.GroupVersionKind) (met bool, statuses []v1alpha1.RequirementStatus, err error) {
 	met = true
 
-	// Check for CRDs
+	// Checking the existence of owned/required CRDs by their Name.
 	for _, r := range crdDescs {
+		g := strings.SplitN(r.Name, ".", 2)
+		var group string
+		if len(g) != 2 {
+			met = false
+			err = fmt.Errorf("CRD name %s is not in the form of `pural.group`", r.Name)
+			return
+		}
+		group = g[1]
 		status := v1alpha1.RequirementStatus{
-			Group:   "apiextensions.k8s.io",
-			Version: "v1beta1",
-			Kind:    "CustomResourceDefinition",
+			Group:   group,
+			Version: r.Version,
+			Kind:    r.Kind,
 			Name:    r.Name,
 		}
 
-		// check if CRD exists - this verifies group, version, and kind, so no need for GVK check via discovery
-		crd, err := a.lister.APIExtensionsV1beta1().CustomResourceDefinitionLister().Get(r.Name)
-		if err != nil {
+		if err := a.isGVKRegistered(group, r.Version, r.Kind); err != nil {
```

Comment:
Is there a reason we're not just building one big combined list of GVKs and checking them all at once (on line 131)?